Source/ASTextNode2.mm

```diff
@@ -323,10 +320,17 @@ - (NSArray *)exclusionPaths
   return _textContainer.exclusionPaths;
 }
 
-- (void)prepareAttributedString:(NSMutableAttributedString *)attributedString
+- (NSAttributedString *)l_preparedAttributedText
 {
-  ASLockScopeSelf();
+  if (_preparedAttributedText) {
+    return _preparedAttributedText;
+  }
+  
+  if (!_attributedText) {
+    return [[NSAttributedString alloc] init];
+  }
  
+  NSMutableAttributedString *attributedString = [_attributedText mutableCopy];
```

Comment:
Why was there not a copy here before? Is it definitely needed here, or should we be doing it at a higher level like setAttributedString?